lib/vacuum/prune/pack/pack.go

```diff
@@ -150,124 +140,219 @@ func (r *cleanup) mark() (required packageMap, err error) {
+				if err != nil {
+					return nil, trace.Wrap(err)
+				}
+
+				if deletePackage {
+					state[item.Locator] = item
+				}
+			}
 		}
 	}
+	return state, nil
+}
 
-	if isAppResourcesPackage(envelope) {
-		appPackage, err := packageForResources(envelope, required)
-		if err != nil {
-			return false, trace.Wrap(err)
-		}
-		appVersion, err := appPackage.Locator.SemVer()
-		if err != nil {
-			return false, trace.Wrap(err)
-		}
-		if version.Compare(*appVersion) < 0 {
-			log.Debug("Will delete an obsolete application resources package.")
-			return true, nil
-		}
+func (r *cleanup) deletePackage(item statePackage) error {
+	r.PrintStep("Deleting package %v.", item.Locator)
+	if r.DryRun {
+		return nil
+	}
+	err := r.Packages.DeletePackage(item.Locator)
+	if err != nil {
+		return trace.Wrap(err)
 	}
+	return nil
+}
 
-	if loc.IsLegacyRuntimePackage(envelope.Locator) {
+// shouldDeletePackage determines if the specified package pkg is eligible for removal.
+// It will match the package against the specified map of required packages.
+// It will also apply a couple of additional ad-hoc heuristics to decide if a package
+// should be deleted
+func (r *cleanup) shouldDeletePackage(pkg existingPackage, required packageMap) (delete bool, err error) {
+	log := r.WithField("package", pkg.Locator)
+
+	if isLegacyRuntimePackage(pkg.PackageEnvelope) {
 		log.Debug("Will delete a legacy runtime package.")
 		return true, nil
 	}
 
-	if isRPCUpdateCredentialsPackage(envelope) && version.Compare(r.runtimeVersion) < 0 {
+	if isRPCUpdateCredentialsPackage(pkg.PackageEnvelope) && pkg.Version.Compare(r.runtimeVersion) < 0 {
 		// Remove obsolete update RPC credentials used in prior update operations.
 		// All RPC packages with versions prior or equal to the currently
 		// installed runtime version are eligible for removal.
 		log.Debug("Will delete an obsolete RPC credentials package.")
 		return true, nil
 	}
 
-	if existingVersion, exists := required[envelope.Locator.ZeroVersion()]; exists {
-		if existingVersion.Compare(*version) > 0 {
+	if existingVersion, exists := required[pkg.Locator.ZeroVersion()]; exists {
+		if existingVersion.Compare(pkg.Version) > 0 {
 			log.Debug("Will delete an obsolete package.")
 			return true, nil
 		}
 		log.Debug("Will not delete a package still in use.")
 	}
 
-	if envelope.Locator.Repository != defaults.SystemAccountOrg {
+	if pkg.Locator.Repository != defaults.SystemAccountOrg {
 		log.Debug("Will not delete from a custom repository.")
 		return false, nil
 	}
 
 	return false, nil
 }
 
-func packageForConfig(envelope pack.PackageEnvelope, packages packageMap) (*requiredPackage, error) {
-	parentPackageName := envelope.RuntimeLabels[pack.ConfigLabel]
-	parentPackageFilter, err := loc.ParseLocator(parentPackageName)
-	if err != nil {
-		return nil, trace.Wrap(err, "invalid package locator")
+// withDependencies computes depender packages for the specified package pkg using
```

Comment:
depender -> dependent?

Reply:
No, it's depender package - i.e. that's the owner of this package, so to speak. The depender package would be the application package for `<app-name>-resources:<app-version>` or `planet-master:<version>` for the `planet-config-<some-suffix>:<version>`